src/segment_reshape/map_tool/segment_reshape_tool.py

```diff
@@ -158,6 +166,21 @@ def _handle_mouse_click_event(
                 success=True,
             )
 
+    def _handle_key_event(self, key: Qt.Key) -> None:
+        if self.reshape_mode is True:
+            if key == Qt.Key_Escape or len(self.added_points) == 0:
+                self.new_segment_rubber_band.reset()
+                self.added_points = []
+                self.temporary_new_segment_rubber_band.reset()
+                self._change_to_pick_location_mode()
+                return
+            elif key in (Qt.Key_Backspace, Qt.Key_Delete):
+                if len(self.added_points) > 1:
```

Comment:
could it be possible to use the `new_segment_rubber_band` instead of keeping a separate `added_points` list? i.e. check for `self.new_segment_rubber_band.numberOfVertices()` here, remove last of it, and if theres still something left update the temp segment first point to the last point of new segment instead?

Reply:
Probably yes. Good idea! I will modify the code to utilize numberOfVertices() function